src/utils.js

```diff
@@ -21,6 +21,19 @@ settings.inUse = settings['force'] || !settings.hasNativeShadowDOM;
 settings.noPatch = settings['noPatch'] || false;
 settings.preferPerformance = settings['preferPerformance'];
 
+const IS_IE = navigator.userAgent.match('Trident');
+settings.IS_IE = IS_IE;
+
+let _canUpgrade;
+// do not use `upgrade` when custom elements is polyfilled or on IE.
```

Comment:
Why not?

Reply:
Elminiate the bail-out on CE polyfill by detecting `shadowRoot.appendChild(shadowRoot)` and doing the `attachDom` work there.

Reply:
IE blows up if we patch the documentFragment.

Optimizations were moved in situ: `attachDOM` puts the fragment on the host and `attachShadow` and `appendChild` use that.